public/app/themes/clarity/inc/admin/plugins/wp-document-revisions.php

```diff
@@ -172,4 +176,102 @@ public function retryFilesNotFound($file, $attachment_id)
+     * @param int|null $post_id The post ID.
+     * @return int|string The author ID or display name.
+     */
+    public function getMostRecentRevisionAuthor(string $format = 'id', int|null $post_id = null): int|string
+    {
+        // Only allow 'id' or 'displayname' as formats.
+        if (!in_array($format, ['id', 'displayname'])) {
+            return 0;
+        }
+
+        // If we don't have a post ID, but we are in the loop, get the post ID.
+        if (!$post_id && get_the_ID()) {
+            $post_id = get_the_ID();
+        }
+
+        // If we haven't already set the wp_document_revisions object, do so now.
+        if (!$this->wp_document_revisions) {
+            $this->wp_document_revisions = new \WP_Document_Revisions();
+        }
+
+        // Get the revisions for the current post - the first in the array is the document, technically not a revision.
+        $document_revisions = $this->wp_document_revisions->get_revisions(get_the_ID());
+
+        // In an edge case we might not have any revisions. If so, return 0.
+        if (empty($document_revisions)) {
+            return 0;
+        }
+
+        // Get the author of the most recent revision.
+        $most_recent_revision_author = $document_revisions[1]->post_author ?? null;
+
+        // If we can't find the author, return the 0;
+        if (!$most_recent_revision_author || !is_numeric($most_recent_revision_author)) {
+            return 0;
+        }
+
+        // Return the author ID or display name.
+        if ($format === 'id') {
+            return (int) $most_recent_revision_author;
+        }
+
+        return get_user_by('ID', $most_recent_revision_author)?->display_name ?? 0;
+    }
+
+    /**
+     * Filter the get_revisions result to correct the author.
+     *
+     * When this filter is called in the context of 'revision_metabox',
+     * it is used to correct the author on the first row of the revisions table.
+     *
+     * @param array $revisions The revisions.
+     * @param string $context The context.
+     * @return array The filtered revisions.
+     */
+    public function filterGetMostRecentRevision($revisions, $context)
+    {
+        if ('revision_metabox' !== $context) {
+            return $revisions;
+        }
+
+        if (empty($revisions) || !is_array($revisions) || !isset($revisions[1])) {
+            return $revisions;
+        }
+
+        $revisions[0]->post_author = $revisions[1]->post_author;
+
+        return $revisions;
+    }
+
+    /**
+     * Filter the get_latest_revision result to correct the author.
+     * 
+     * When this filter is called in the context of 'document_metabox',
+     * it is used to correct the author in the string 'Checked in x ago by y'.
+     * 
+     * @param mixed $revision The revision.
+     * @param string $context The context.
+     * @return mixed The filtered revision.
+     */
+    public function filterGetLatestRevision(mixed $revision, string $context): mixed
```

Comment:
Instead of mixed, could the $revision param type be WP_Post here?

Reply:
Thanks, I've used WP_Post in the docblock.

To reflect the code in wp-document-revisions, and to play it safe... I've removed the native php types - as if they are wrong, php will throw a fatal error.